### PR TITLE
Use the last revision of the branch's remote on API requests

### PR DESCRIPTION
**Pull request by kbolashev**
Now on each API request the client is opening two files:
- .git/HEAD to see what branch/revision we're on
- the HEAD of the remote for the associated branch

I'm not sure if trying to do some logic around caching them is a good idea

### CODE REVIEW

**Review by simonlsk — CHANGES_REQUESTED** [review]

File: dagshub/streaming/filesystem.py [hunk 1]
```diff
@@ -167,23 +201,28 @@ def auth(self):
...
             if remote.hostname != config.hostname:
                 continue
             remote = remote._replace(netloc=remote.hostname)
             remote = remote._replace(path=re.compile(r'(\.git)?/?$').sub('', remote.path))
-            dagshub_remotes.append(remote.geturl())
-
-        return dagshub_remotes
+            self.dagshub_remotes.append(remote.geturl())
+        # Build a dictionary of branch: path to remote's HEAD
+        for branch, branch_section in [(b, git_config[b]) for b in git_config if b.startswith("branch ")]:
```
[the remark this answers is not in the document]
Comment (simonlsk) [reply dagshub/streaming/filesystem.py]: not relevant anymore

File: dagshub/streaming/filesystem.py [hunk 2]
```diff
@@ -517,6 +556,10 @@ def __repr__(self):
 
 # Used for testing purposes only
 if __name__ == "__main__":
-    install_hooks()
+    logging.basicConfig(level=logging.DEBUG)
+    if len(sys.argv) > 1:
+        os.chdir(sys.argv[1])
+    fs = DagsHubFilesystem()
+    fs.install_hooks()
```
[the remark this answers is not in the document]
Comment (kbolashev) [reply dagshub/streaming/filesystem.py:561]: Local debugging. So I can install_hooks on a repo by providing a path to it in argv.

**Review by simonlsk — APPROVED** [review]

**Comment by simonlsk** [comment]
Linter is not passing.

**Comment by kbolashev** [comment]
Oops, sorry for that and thanks for noticing

**Pull request merged**